flink-tests/src/test/java/org/apache/flink/test/example/client/JobRetrievalITCase.java

```diff
@@ -80,42 +95,38 @@ public void testJobRetrieval() throws Exception {
 
 		final JobGraph jobGraph = new JobGraph(jobID, "testjob", imalock);
 
-		final ClusterClient<StandaloneClusterId> client = new StandaloneClusterClient(cluster.configuration(), cluster.highAvailabilityServices(), true);
-
 		// acquire the lock to make sure that the job cannot complete until the job client
 		// has been attached in resumingThread
 		lock.acquire();
```

Comment:
The locking should no longer be needed in Flip-6 mode. The job result will also be available after the job has finished.

Reply:
Just tested it and it worked.

Reply:
That would no longer verify that you can connect to a _running_  job though.

Reply:
In Flip-6 there is no such concept as attaching to a running job anymore. 

What attaching effectively did in the legacy code, was to tell the `JobManager` that there is again a client waiting to be notified about the job result.

In Flip-6 the job result will be stored on the `Dispatcher` also if there is no client waiting to be notified. This allows the client to request it at any time (modulo some cache size cleaning conditions).

Reply:
But there is still a difference between asking for the result when the job is finished (where you get it right away) versus when the job running (where you have to poll).

The test should cover both.

Reply:
Code-wise it doesn't really make a difference for Flip-6. Given that the job takes a little bit of time, then we should fuzzily cover both cases when removing the lock, though.